src/frontend/cellprofiler/modules/measurecolocalization.py

```diff
@@ -356,21 +594,177 @@ def run(self, workspace):
+        # Thresholding code used from run_image_pair_images() and run_image_pair_objects()
+        image_pixel_data = image.pixel_data
+        image_mask = image.mask
+        image_mask = image_mask & (~numpy.isnan(image_pixel_data))
+        output_image_arr = numpy.zeros_like(image_pixel_data)
+        if object_name is None:
+            # perform on the entire image
+            if numpy.any(image_mask):
+                    thr_i = t_val * numpy.max(image_pixel_data) / 100
+                    output_image_arr = image_pixel_data > thr_i
+        else:
+            # perform on the object
+            objects = workspace.object_set.get_objects(object_name)
+            labels = objects.segmented
+            try:
+                image_pixels = objects.crop_image_similarly(image.pixel_data)
+                image_mask = objects.crop_image_similarly(image.mask)
+            except ValueError:
+                image_pixels, m1 = size_similarly(labels, image.pixel_data)
+                image_mask, m1 = size_similarly(labels, image.mask)
+                image_mask[~m1] = False
+
+            mask = ((labels > 0) & image_mask) & (~numpy.isnan(image_pixels))
+            labels = labels[mask]
+            
+            if numpy.any(mask):
+                image_pixels = image_pixels[mask]
+            n_objects = objects.count
+
+            if (not (n_objects == 0)) and (not (numpy.where(mask)[0].__len__() == 0)):
+                lrange = numpy.arange(n_objects, dtype=numpy.int32) + 1
+                # Threshold as percentage of maximum intensity of objects in each channel
+                scaled_image = (t_val / 100) * fix(
+                    scipy.ndimage.maximum(image_pixels, labels, lrange)
+                )
+
+                # convert 1d array into 2d image using mask as index
+                output_image_arr = numpy.zeros_like(mask)
+                output_image_arr[mask] = (image_pixels >= scaled_image[labels - 1])
+        return output_image_arr
+
+
+    def save_requested_masks(self, workspace):
+        # Iterate over the list of save masks
+        for save_mask in self.save_mask_list:
+            image_name = save_mask.image_name.value
+            object_name = save_mask.choose_object.value if save_mask.save_mask_wants_objects.value else None
+            save_image_name = save_mask.save_image_name.value
+            original_image = workspace.image_set.get_image(image_name, must_be_grayscale=True)
+            
+            # Call the relevant funcitons to get the thresholded masks
+            output_image = Image(self.get_thresholded_mask(workspace, image_name, object_name), parent_image=original_image)
+
+            # Save the mask to the image set
+            workspace.image_set.add(save_image_name, output_image)
+            
+
+
+    def get_image_threshold_value(self, image_name):
+        if self.wants_channel_thresholds.value:
+            for threshold in self.thresholds_list:
```

Comment:
Not using a dict here to avoid writing extra logic for keeping the dict up to date with the list